common/core/desktop/src/state.cpp

```diff
@@ -26,6 +27,17 @@ void actions::push_persist(option const &&opt) {
   emplace_back(std::move(ai));
 }
 
+void actions::push_capslock(bool turnOn)
+{
+  assert(empty() || (!empty() && back().type != KM_KBP_IT_END));
+  option option = {KM_KBP_OPT_ENVIRONMENT, KM_KBP_KMX_ENV_CAPSLOCK, turnOn ? u"1" : u"0"};
```

Comment:
Does this look right? We got rid of capsLock in the environment, but this is a different env/options. Or is there a better way to do this?

Reply:
No, I don't think we want to be adding it as an env option: rather, we want to pass the on/off flag directly in the `KM_KBP_IT_CAPSLOCK` action. Otherwise we end up with the flag being visible in two places, which is just messy, and also passing strings for an internal boolean flag which is equally ugly!

I suggest we add `uint8_t capsLock; // CAPSLOCK type, 1 to turn on, 0 to turn off` to the union in `km_kbp_action_item`.

Reply:
sounds good. done.